### PR TITLE
PR for test e2e within pipeline

### PR DESCRIPTION
**Pull request by OneHedgehog**
PR which covers e2e for the VScode extension

### CODE REVIEW

File: src/test/helpers/authentication.ts [hunk 1]
```diff
@@ -61,7 +61,9 @@ export class FakeAuthenticationProviderManager {
    */
   async getSession(
     providerId: string,
-    scopes: readonly string[],
+    scopeListOrRequest:
```
Comment (OneHedgehog) [inline src/test/helpers/authentication.ts:64]: This required after updating node_modules, as now "scopeListOrRequest" has also  "vscode.AuthenticationWwwAuthenticateRequest" (before it was only string[]

File: src/test/extension.e2e.test.ts [hunk 2]
```diff
@@ -58,23 +57,31 @@ describe('Colab Extension', function () {
...
 
-    it('authenticates and executes the notebook on a Colab server', async () => {
+    it('authenticates and executes the notebook on a Workbench server', async () => {
       // Select the Colab server provider from the kernel selector.
       await workbench.executeCommand('Notebook: Select Notebook Kernel');
-      await selectQuickPickItem({
-        item: 'Colab',
-        quickPick: 'Select Another Kernel',
+      const selected = await selectAnyQuickPickItem({
+        items: ['Select Another Kernel...', 'Google Cloud Workbench'],
```
Comment (OneHedgehog) [inline src/test/extension.e2e.test.ts:70]: Depending of environment, user could either see all options of menu from the start (clean env) OR to see his previously selected kernel (dirty env)

We need to cover both cases

File: src/test/extension.e2e.test.ts [hunk 3]
```diff
@@ -287,10 +314,28 @@ async function waitAndClick(
...
+          until.elementIsVisible(element),
+          ELEMENT_WAIT_MS,
+          `Element located but not visible: ${errorMsg}`,
+        );
+        await element.click();
+        return true;
+      } catch (e: unknown) {
+        if (
+          e instanceof Error &&
+          (e.name === 'StaleElementReferenceError' ||
```
Comment (OneHedgehog) [inline src/test/extension.e2e.test.ts:336]: retry, if element us stale or click intercepted.

File: .github/workflows/build-test.yml [hunk 4]
```diff
@@ -23,14 +23,17 @@ on:
...
 jobs:
   generateConfig:
     name: Generate Config
     runs-on: ubuntu-latest
+    environment: Workbench
     env:
-      WORKBENCH_EXTENSION_CLIENT_ID: "client-id"
-      WORKBENCH_EXTENSION_CLIENT_NOT_SO_SECRET: "client-not-so-secret"
+      WORKBENCH_EXTENSION_CLIENT_ID: ${{ vars.WORKBENCH_EXTENSION_CLIENT_ID }}
+      WORKBENCH_EXTENSION_CLIENT_NOT_SO_SECRET: ${{ vars.WORKBENCH_EXTENSION_CLIENT_NOT_SO_SECRET }}
```
Comment (alexkallaur) [inline .github/workflows/build-test.yml]: Few comments here.
1) AFAIK, all sensitive info: creds, secrets, etc should go into Environment Secrets, not Environment Variables. Variables might be visible in GH Actions logs, which means we probably now have to check it. Likely we will need to purge them and/or maybe changing the PWD for the test user as a minimum.
2) There are COLAB_* leftovers in Secrets, PTAL.
3) Node version in Variables does not match the one you have here in this file. Do we need both? which one should we use?

File: src/test/helpers/authentication.ts
Comment (alexkallaur) [inline src/test/helpers/authentication.ts:1]: Could you please clarify why we need these changes in the file for a fake impl?

File: src/test/extension.e2e.test.ts [hunk 5]
```diff
@@ -19,32 +19,31 @@ import {
   VSBrowser,
   until,
 } from 'vscode-extension-tester';
+import { CONFIG } from '../config';
 
-const ELEMENT_WAIT_MS = 10000;
+const ELEMENT_WAIT_MS = 60000;
```
Comment (alexkallaur) [inline src/test/extension.e2e.test.ts]: nit: use 60 * 1000, that would make it easy to parse and understand that it is 1 min.

File: src/test/extension.e2e.test.ts [hunk 6]
```diff
@@ -19,32 +19,31 @@ import {
...
-    assert.equal(
-      'production',
-      'Unexpected extension environment. Run `npm run generate:config`',
-    );
+    assert.ok(CONFIG.ClientId, 'ClientId is not set');
+    assert.ok(CONFIG.ClientNotSoSecret, 'ClientNotSoSecret is not set');
     // Wait for VS Code UI to settle before running tests.
     workbench = new Workbench();
     driver = workbench.getDriver();
     await driver.sleep(8000);
```
Comment (alexkallaur) [inline src/test/extension.e2e.test.ts]: those `sleep` values look a bit chaotic. Can we have a dedicated constant for them, similar to `ELEMENT_WAIT_MS`. Also, it is not clear why this one is 8 times bigger than the one on line 97. Consider leaving a comment explaining.

File: src/test/extension.e2e.test.ts [hunk 7]
```diff
@@ -86,22 +93,32 @@ describe('Colab Extension', function () {
...
       await inputBox.sendKeys(Key.ENTER);
       await selectQuickPickItem({
-        item: 'Python 3 (ipykernel)',
-        quickPick: 'Select a Kernel from Colab CPU',
+        item: 'workbench-vs-code-plugin (jaas-test-notebooks-host)',
+        quickPick: 'Select a Jupyter Server',
+      });
+
+      await selectQuickPickItem({
+        item: 'TensorFlow 2-11',
```
Comment (alexkallaur) [inline src/test/extension.e2e.test.ts:123]: Why not `Python 3`?

File: eslint.config.mts [hunk 8]
```diff
@@ -16,6 +16,8 @@ import tsDocPlugin from 'eslint-plugin-tsdoc';
 import globals from 'globals';
 import tseslint from 'typescript-eslint';
 
+// TODO need to update config
+// eslint-disable-next-line @typescript-eslint/no-deprecated
```
Comment (alexkallaur) [inline eslint.config.mts:20]: Does it need to be addressed?

**Review by alexkallaur — CHANGES_REQUESTED** [review]
Left a few comments. PTAL

File: .github/workflows/build-test.yml [hunk 9]
```diff
@@ -23,14 +23,17 @@ on:
...
   # See supported Node.js release schedule at https://nodejs.org/en/about/releases
   NODE_VERSION: 22.x
+  TEST_ACCOUNT_EMAIL: ${{ vars.TEST_ACCOUNT_EMAIL }}
+  TEST_ACCOUNT_PASSWORD: ${{ vars.TEST_ACCOUNT_PASSWORD }}
 
 jobs:
   generateConfig:
     name: Generate Config
     runs-on: ubuntu-latest
+    environment: Workbench
```
Comment (alexkallaur) [inline .github/workflows/build-test.yml:33]: in .env.template I see
`# One of "production", "sandbox", "local"`
Either we changes this name or the comment in the template

File: src/test/helpers/authentication.ts
Comment (OneHedgehog) [reply src/test/helpers/authentication.ts:1]: Added a comment

File: eslint.config.mts (hunk 8 above)
Comment (OneHedgehog) [reply eslint.config.mts:20]: Yes, but I'd like to do in separate PR. Actually, in next PR (to not mix-up things here)

File: src/test/extension.e2e.test.ts [hunk 10]
```diff
@@ -58,23 +57,31 @@ describe('Colab Extension', function () {
       await cell.sendKeys('1 + 1');
```
Comment (hjjackyang) [inline src/test/extension.e2e.test.ts:63]: Nit (pre-existing): Something we didn't do on Consumer side but should probably consider doing is to move the test data into a separate ipynb file, and then feed that file into E2E tests.

Not sure how easy or hard it is to do it. Something to consider in the future.

**Review by hjjackyang — COMMENTED** [review]
No real concern from me other than 1 non-actionable comment + all the comments Alex left. I will leave the approval to Alex.

File: .github/workflows/build-test.yml (hunk 9 above)
Comment (OneHedgehog) [reply .github/workflows/build-test.yml:33]: We don't use it. I'll remove the comment

File: src/test/extension.e2e.test.ts (hunk 7 above)
Comment (OneHedgehog) [reply src/test/extension.e2e.test.ts:123]: python3 kernel in it's name will also have a previous selected notebook name for local setup. For clean env (pipeline) this is not the case. So for now, I've done this for simplicity

File: .github/workflows/build-test.yml (hunk 4 above)
Comment (OneHedgehog) [reply .github/workflows/build-test.yml]: For the sensitive env, initially, I've done that, as github free doesn't allow usage of secrets for a private repo. However, this organisation (Google Cloud) has subscription, which make this possible, so done

File: src/test/extension.e2e.test.ts (hunk 5 above)
Comment (OneHedgehog) [reply src/test/extension.e2e.test.ts]: done, thank you

File: src/test/extension.e2e.test.ts (hunk 6 above)
Comment (OneHedgehog) [reply src/test/extension.e2e.test.ts]: done, thank you

File: src/test/extension.e2e.test.ts [hunk 11]
```diff
@@ -58,23 +63,31 @@ describe('Colab Extension', function () {
...
     });
 
-    it('authenticates and executes the notebook on a Colab server', async () => {
+    it('authenticates and executes the notebook on a Workbench server', async () => {
       // Select the Colab server provider from the kernel selector.
```
Comment (alexkallaur) [inline src/test/extension.e2e.test.ts]: nit: "Colab" -> "Workbench"

**Review by alexkallaur — APPROVED** [review]
LGTM with 1 nit

Please, do not forget to create a follow-up bug to update the linter specific TODO item in eslint.config.mts.

File: src/test/extension.e2e.test.ts (hunk 11 above)
Comment (OneHedgehog) [reply src/test/extension.e2e.test.ts]: Done, thank you

**Comment by OneHedgehog** [comment]
> LGTM with 1 nit
> 
> Please, do not forget to create a follow-up bug to update the linter specific TODO item in eslint.config.mts.

For the eslint, let's wait till this PR get merged and I'll open a new one to fix. The reason for that, is because this issue happens after node_modules updated. So, I want to avoid conflicts in package-lock.json between PRs
